packages/wonder-blocks-core/src/hooks/__tests__/use-is-mounted.test.tsx

```diff
@@ -0,0 +1,102 @@
+import * as React from "react";
+import {renderHook} from "@testing-library/react-hooks";
+import {renderToString} from "react-dom/server";
+
+import Server from "../../util/server";
+
+import {useIsMounted} from "../use-is-mounted";
+
+describe("useIsMounted", () => {
+    it("should return a function", () => {
+        // Arrange
+
+        // Act
+        const {result} = renderHook(useIsMounted);
+
+        // Assert
+        expect(result.current).toBeFunction();
+    });
+
+    describe("when client side, the returned function", () => {
+        beforeEach(() => {
+            jest.spyOn(Server, "isServerSide").mockReturnValue(false);
+        });
+
+        it("should return false on initial render", () => {
+            // Arrange
+            jest.spyOn(console, "error").mockImplementation(() => {
+                // We need to capture the initial render only, without mounting
+                // so we can't use Testing Library. Instead, we're going to
+                // use `renderToString` from `react-dom/server`, but that is
+                // going to log an error because we use `useLayoutEffect`.
+                // So, we suppress that error for our testing purposes.
+                /* no-op */
+            });
+            const Component = () => {
+                const isMounted = useIsMounted();
+                return isMounted() ? <>MOUNTED</> : <>UNMOUNTED</>;
+            };
+
+            // Act
+            const result = renderToString(<Component />);
+
+            // Assert
+            expect(result).toBe("UNMOUNTED");
+        });
```

Comment:
**suggestion (non-blocking):** you could also use `result.all` with `renderHook()` to check all the values that were returned from the hook.

Reply:
Hmm, I could, but this test case feels more succinct in that we know for sure this is the initial render we want.

Reply:
I tried `result.all` but that only had one value and that returned `true`. I don't think that Testing Library can represent that very first, pre-mount server/hydration render.